File: probs/30885.cpp
#include <bits/stdc++.h>

using namespace std;
using ll = long long;
using pi = pair<ll, ll>;

list<pi> L;

int main()
{
    ios_base::sync_with_stdio(0);
    cin.tie(0);

    freopen("input.txt", "r", stdin);

    int N; cin >> N;
    for (int i = 1; i <= N; i++) {
        int a; cin >> a;
        L.emplace_back(a, i);
    }

    while (L.size() > 1) {
        for (auto it = L.begin(); it != L.end(); it++) {
            ll x = it->first;
            if (it != L.begin() && prev(it)->first <= x) {
                it->first += prev(it)->first;
                L.erase(prev(it));
            }
            if (next(it) != L.end() && next(it)->first <= x) {
                it->first += next(it)->first;
                L.erase(next(it));
            }
        }
    }

    cout << L.begin()->first << '\n' << L.begin()->second << '\n';

    return 0;
}

File: probs/30885.py
import sys
input = sys.stdin.readline

input = open('input.txt', 'r').readline

class Node:
    def __init__(self, n, i, p=None):
        self.n = n
        self.i = i
        self.prev = p
        self.next = None

    def __str__(self):
        return f'{self.n}\n{self.i}'

N = int(input())
L = cur = Node(0, 0)
for i, n in enumerate(map(int, input().split()), 1):
    tmp = Node(n, i, cur)
    cur.next = tmp
    cur = tmp
cur.next = L
L.prev = cur

while L.next.next is not L:
    cur = L.next
    while cur is not L:
        n = cur.n
        if cur.prev is not None and cur.prev is not L and cur.prev.n <= n:
            cur.n += cur.prev.n
            cur.prev = cur.prev.prev
            cur.prev.next = cur
        if cur.next is not None and cur.next is not L and cur.next.n <= n:
            cur.n += cur.next.n
            cur.next = cur.next.next
            cur.next.prev = cur
        cur = cur.next

print(L.next)
